# **Checkpoint 2**

In [ ]:
import pandas as pd

file_path ="/content/drive/MyDrive/praktikum avd/BMIData.csv"

df=pd.read_csv(file_path)

df.head()


SchoolYear NameHospital     Sex  EpiUnderweight  EpiHealthyWeight  \
0        2001      Borders  Female               2               334   
1        2001      Borders    Male               5               301   
2        2001      Lothian  Female              12               728   
3        2001      Lothian    Male              17               681   
4        2001         Fife  Female              17              1361   

   EpiOverweight  EpiObese  EpiOverweightAndObese  ValidCounts  UnvalidCounts  \
0             54        28                     82          418             82   
1             63        42                    105          411            105   
2            133       106                    239          979            239   
3            124       114                    238          936            238   
4            231       155                    386         1764            386   

   TotalCounts  
0          500  
1          516  
2         1218  
3         1174  
4         2150

In [ ]:
import pandas as pd

# **Data Understanding**


Dataset ini diperoleh dari Kaggle-klasifikasi berat badan anak.

In [ ]:
df = pd.read_csv("/content/drive/MyDrive/praktikum avd/BMIData.csv")


**1. Cek jumlah baris dan kolom**

Dataset memiliki 516 baris dan 11 kolom.

In [ ]:
print("Jumlah baris dan kolom:", df.shape)

Jumlah baris dan kolom: (516, 11)


**2. Cek tipe data setiap kolom**

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 516 entries, 0 to 515
Data columns (total 11 columns):
 #   Column                 Non-Null Count  Dtype 
---  ------                 --------------  ----- 
 0   SchoolYear             516 non-null    int64 
 1   NameHospital           516 non-null    object
 2   Sex                    516 non-null    object
 3   EpiUnderweight         516 non-null    int64 
 4   EpiHealthyWeight       516 non-null    int64 
 5   EpiOverweight          516 non-null    int64 
 6   EpiObese               516 non-null    int64 
 7   EpiOverweightAndObese  516 non-null    int64 
 8   ValidCounts            516 non-null    int64 
 9   UnvalidCounts          516 non-null    int64 
 10  TotalCounts            516 non-null    int64 
dtypes: int64(9), object(2)
memory usage: 44.5+ KB


**3. Tampilkan 5 data pertama dan terakhir untuk melihat isi dataset (jika baris banyak)**

In [ ]:
df


SchoolYear               NameHospital     Sex  EpiUnderweight  \
0          2001                    Borders  Female               2   
1          2001                    Borders    Male               5   
2          2001                    Lothian  Female              12   
3          2001                    Lothian    Male              17   
4          2001                       Fife  Female              17   
..          ...                        ...     ...             ...   
511        2023                    Tayside    Male              15   
512        2023  Greater Glasgow and Clyde  Female              34   
513        2023  Greater Glasgow and Clyde    Male              89   
514        2023                Lanarkshire  Female              14   
515        2023                Lanarkshire    Male              31   

     EpiHealthyWeight  EpiOverweight  EpiObese  EpiOverweightAndObese  \
0                 334             54        28                     82   
1                 301             63        42                    105   
2                 728            133       106                    239   
3                 681            124       114                    238   
4                1361            231       155                    386   
..                ...            ...       ...                    ...   
511              1201            216       178                    394   
512              4097            595       561                   1156   
513              4152            615       641                   1256   
514              2255            381       330                    711   
515              2280            348       381                    729   

     ValidCounts  UnvalidCounts  TotalCounts  
0            418             82          500  
1            411            105          516  
2            979            239         1218  
3            936            238         1174  
4           1764            386         2150  
..           ...            ...          ...  
511         1610            394         2004  
512         5287           1156         6443  
513         5497           1256         6753  
514         2980            711         3691  
515         3040            729         3769  

[516 rows x 11 columns]

**4. statistik deskriptif**

In [ ]:
df.describe()

SchoolYear  EpiUnderweight  EpiHealthyWeight  EpiOverweight  \
count   516.000000      516.000000         516.00000     516.000000   
mean   2013.383721       21.742248        1304.72093     208.102713   
std       5.947672       25.207054        1103.90967     167.192486   
min    2001.000000        0.000000           7.00000       0.000000   
25%    2009.000000        3.000000         421.75000      66.750000   
50%    2013.500000       13.000000        1158.50000     196.000000   
75%    2018.000000       30.000000        1946.25000     292.000000   
max    2023.000000      137.000000        4623.00000     696.000000   

         EpiObese  EpiOverweightAndObese  ValidCounts  UnvalidCounts  \
count  516.000000             516.000000   516.000000     516.000000   
mean   175.563953             383.666667  1710.129845     383.666667   
std    147.842378             313.523071  1433.191567     313.523071   
min      2.000000               2.000000    10.000000       2.000000   
25%     52.000000             120.500000   549.000000     120.500000   
50%    160.500000             350.500000  1558.500000     350.500000   
75%    243.000000             537.000000  2523.250000     537.000000   
max    737.000000            1433.000000  6150.000000    1433.000000   

       TotalCounts  
count   516.000000  
mean   2093.796512  
std    1743.889989  
min      12.000000  
25%     669.500000  
50%    1904.500000  
75%    3038.750000  
max    7583.000000

## **Data Preparation**

Dataset ini diperoleh dari Kaggle-klasifikasi berat badan anak.

# **A.Pengecekan Terhadap Data**

**1. nilai yang hilang (Missing Values)**

In [ ]:
print((df.isna().sum() / len(df)) * 100)

SchoolYear               0.0
NameHospital             0.0
Sex                      0.0
EpiUnderweight           0.0
EpiHealthyWeight         0.0
EpiOverweight            0.0
EpiObese                 0.0
EpiOverweightAndObese    0.0
ValidCounts              0.0
UnvalidCounts            0.0
TotalCounts              0.0
dtype: float64


**2. Duplicated Values**

In [ ]:
duplicate_rows = df[df.duplicated()]
print("\nJumlah Baris Duplikat:", len(duplicate_rows))


Jumlah Baris Duplikat: 0


**3. Outliers**

In [ ]:
results = []

cols = df.select_dtypes(include=['float64', 'int64'])

for col in cols:
  q1 = df[col].quantile(0.25)
  q3 = df[col].quantile(0.75)
  iqr = q3 - q1
  lower_bound = q1 - 1.5*iqr
  upper_bound = q3 + 1.5*iqr
  outliers = df[(df[col] < lower_bound) | (df[col] > upper_bound)]
  percent_outliers = (len(outliers)/len(df))*100
  results.append({'Kolom': col, 'Persentase Outliers': percent_outliers})

# Dataframe dari list hasil
results_df = pd.DataFrame(results)
results_df.set_index('Kolom', inplace=True)
results_df = results_df.rename_axis(None, axis=0).rename_axis('Kolom', axis=1)

# Tampilkan dataframe
display(results_df)

Kolom                  Persentase Outliers
SchoolYear                        0.000000
EpiUnderweight                    6.589147
EpiHealthyWeight                  2.713178
EpiOverweight                     2.713178
EpiObese                          3.682171
EpiOverweightAndObese             2.713178
ValidCounts                       2.906977
UnvalidCounts                     2.713178
TotalCounts                       2.906977

In [ ]:
df

SchoolYear               NameHospital     Sex  EpiUnderweight  \
0          2001                    Borders  Female               2   
1          2001                    Borders    Male               5   
2          2001                    Lothian  Female              12   
3          2001                    Lothian    Male              17   
4          2001                       Fife  Female              17   
..          ...                        ...     ...             ...   
511        2023                    Tayside    Male              15   
512        2023  Greater Glasgow and Clyde  Female              34   
513        2023  Greater Glasgow and Clyde    Male              89   
514        2023                Lanarkshire  Female              14   
515        2023                Lanarkshire    Male              31   

     EpiHealthyWeight  EpiOverweight  EpiObese  EpiOverweightAndObese  \
0                 334             54        28                     82   
1                 301             63        42                    105   
2                 728            133       106                    239   
3                 681            124       114                    238   
4                1361            231       155                    386   
..                ...            ...       ...                    ...   
511              1201            216       178                    394   
512              4097            595       561                   1156   
513              4152            615       641                   1256   
514              2255            381       330                    711   
515              2280            348       381                    729   

     ValidCounts  UnvalidCounts  TotalCounts  
0            418             82          500  
1            411            105          516  
2            979            239         1218  
3            936            238         1174  
4           1764            386         2150  
..           ...            ...          ...  
511         1610            394         2004  
512         5287           1156         6443  
513         5497           1256         6753  
514         2980            711         3691  
515         3040            729         3769  

[516 rows x 11 columns]

**4. Incosistent Value**

Menurut Saya, dataset yang saya teliti tidak ada nilai yang tidak konsisten

## **B. Penanganan terhadap Data**

**1. nilai yang hilang (Missing Values)**

      tidak ada nilai yang hilang

In [ ]:
missing_values = df.isnull().sum()


**2. Duplicated Values (Dihapus semua duplikasi)**

In [ ]:
df = df.drop_duplicates()

In [ ]:
df

SchoolYear               NameHospital     Sex  EpiUnderweight  \
0          2001                    Borders  Female               2   
1          2001                    Borders    Male               5   
2          2001                    Lothian  Female              12   
3          2001                    Lothian    Male              17   
4          2001                       Fife  Female              17   
..          ...                        ...     ...             ...   
511        2023                    Tayside    Male              15   
512        2023  Greater Glasgow and Clyde  Female              34   
513        2023  Greater Glasgow and Clyde    Male              89   
514        2023                Lanarkshire  Female              14   
515        2023                Lanarkshire    Male              31   

     EpiHealthyWeight  EpiOverweight  EpiObese  EpiOverweightAndObese  \
0                 334             54        28                     82   
1                 301             63        42                    105   
2                 728            133       106                    239   
3                 681            124       114                    238   
4                1361            231       155                    386   
..                ...            ...       ...                    ...   
511              1201            216       178                    394   
512              4097            595       561                   1156   
513              4152            615       641                   1256   
514              2255            381       330                    711   
515              2280            348       381                    729   

     ValidCounts  UnvalidCounts  TotalCounts  
0            418             82          500  
1            411            105          516  
2            979            239         1218  
3            936            238         1174  
4           1764            386         2150  
..           ...            ...          ...  
511         1610            394         2004  
512         5287           1156         6443  
513         5497           1256         6753  
514         2980            711         3691  
515         3040            729         3769  

[516 rows x 11 columns]

**3. Outliers (Melakukan imputasi)**

In [ ]:
def impute_outliers(col):
    median = col.median()
    return col.apply(lambda x: median if x < lower_bound[col.name] or x > upper_bound[col.name] else x)


Melakukan cek ulang

In [ ]:
results = []

cols = df.select_dtypes(include=['float64', 'int64'])

for col in cols:
  q1 = df[col].quantile(0.25)
  q3 = df[col].quantile(0.75)
  iqr = q3 - q1
  lower_bound = q1 - 1.5*iqr
  upper_bound = q3 + 1.5*iqr
  outliers = df[(df[col] < lower_bound) | (df[col] > upper_bound)]
  percent_outliers = (len(outliers)/len(df))*100
  results.append({'Kolom': col, 'Persentase Outliers': percent_outliers})

# Dataframe dari list hasil
results_df = pd.DataFrame(results)
results_df.set_index('Kolom', inplace=True)
results_df = results_df.rename_axis(None, axis=0).rename_axis('Kolom', axis=1)

# Tampilkan dataframe
display(results_df)

Kolom                  Persentase Outliers
SchoolYear                        0.000000
EpiUnderweight                    6.589147
EpiHealthyWeight                  2.713178
EpiOverweight                     2.713178
EpiObese                          3.682171
EpiOverweightAndObese             2.713178
ValidCounts                       2.906977
UnvalidCounts                     2.713178
TotalCounts                       2.906977

In [ ]:
df

SchoolYear               NameHospital     Sex  EpiUnderweight  \
0          2001                    Borders  Female               2   
1          2001                    Borders    Male               5   
2          2001                    Lothian  Female              12   
3          2001                    Lothian    Male              17   
4          2001                       Fife  Female              17   
..          ...                        ...     ...             ...   
511        2023                    Tayside    Male              15   
512        2023  Greater Glasgow and Clyde  Female              34   
513        2023  Greater Glasgow and Clyde    Male              89   
514        2023                Lanarkshire  Female              14   
515        2023                Lanarkshire    Male              31   

     EpiHealthyWeight  EpiOverweight  EpiObese  EpiOverweightAndObese  \
0                 334             54        28                     82   
1                 301             63        42                    105   
2                 728            133       106                    239   
3                 681            124       114                    238   
4                1361            231       155                    386   
..                ...            ...       ...                    ...   
511              1201            216       178                    394   
512              4097            595       561                   1156   
513              4152            615       641                   1256   
514              2255            381       330                    711   
515              2280            348       381                    729   

     ValidCounts  UnvalidCounts  TotalCounts  
0            418             82          500  
1            411            105          516  
2            979            239         1218  
3            936            238         1174  
4           1764            386         2150  
..           ...            ...          ...  
511         1610            394         2004  
512         5287           1156         6443  
513         5497           1256         6753  
514         2980            711         3691  
515         3040            729         3769  

[516 rows x 11 columns]

**4. Incosistent Value**

Menurut Saya, dataset yang saya teliti tidak ada nilai yang tidak konsisten

# **konstruksi data (Construct Data) (penambahan kolom)**

**Menambahkan kolom Persentase Berat Badan Tidak Sehat**

In [ ]:
df["UnhealthyPercentage"] = ((df["EpiUnderweight"] + df["EpiOverweightAndObese"]) / df["TotalCounts"]) * 100


In [ ]:
df

SchoolYear               NameHospital     Sex  EpiUnderweight  \
0          2001                    Borders  Female               2   
1          2001                    Borders    Male               5   
2          2001                    Lothian  Female              12   
3          2001                    Lothian    Male              17   
4          2001                       Fife  Female              17   
..          ...                        ...     ...             ...   
511        2023                    Tayside    Male              15   
512        2023  Greater Glasgow and Clyde  Female              34   
513        2023  Greater Glasgow and Clyde    Male              89   
514        2023                Lanarkshire  Female              14   
515        2023                Lanarkshire    Male              31   

     EpiHealthyWeight  EpiOverweight  EpiObese  EpiOverweightAndObese  \
0                 334             54        28                     82   
1                 301             63        42                    105   
2                 728            133       106                    239   
3                 681            124       114                    238   
4                1361            231       155                    386   
..                ...            ...       ...                    ...   
511              1201            216       178                    394   
512              4097            595       561                   1156   
513              4152            615       641                   1256   
514              2255            381       330                    711   
515              2280            348       381                    729   

     ValidCounts  UnvalidCounts  TotalCounts  UnhealthyPercentage  
0            418             82          500            16.800000  
1            411            105          516            21.317829  
2            979            239         1218            20.607553  
3            936            238         1174            21.720613  
4           1764            386         2150            18.744186  
..           ...            ...          ...                  ...  
511         1610            394         2004            20.409182  
512         5287           1156         6443            18.469657  
513         5497           1256         6753            19.917074  
514         2980            711         3691            19.642373  
515         3040            729         3769            20.164500  

[516 rows x 12 columns]

**Menambahkan kolom Kategori Risiko Obesitas**

In [ ]:
def categorize_risk(value):
    if value < 10:
        return "Low"
    elif value < 20:
        return "Moderate"
    else:
        return "High"

df["ObesityRiskCategory"] = df["UnhealthyPercentage"].apply(categorize_risk)

In [ ]:
df

SchoolYear               NameHospital     Sex  EpiUnderweight  \
0          2001                    Borders  Female               2   
1          2001                    Borders    Male               5   
2          2001                    Lothian  Female              12   
3          2001                    Lothian    Male              17   
4          2001                       Fife  Female              17   
..          ...                        ...     ...             ...   
511        2023                    Tayside    Male              15   
512        2023  Greater Glasgow and Clyde  Female              34   
513        2023  Greater Glasgow and Clyde    Male              89   
514        2023                Lanarkshire  Female              14   
515        2023                Lanarkshire    Male              31   

     EpiHealthyWeight  EpiOverweight  EpiObese  EpiOverweightAndObese  \
0                 334             54        28                     82   
1                 301             63        42                    105   
2                 728            133       106                    239   
3                 681            124       114                    238   
4                1361            231       155                    386   
..                ...            ...       ...                    ...   
511              1201            216       178                    394   
512              4097            595       561                   1156   
513              4152            615       641                   1256   
514              2255            381       330                    711   
515              2280            348       381                    729   

     ValidCounts  UnvalidCounts  TotalCounts  UnhealthyPercentage  \
0            418             82          500            16.800000   
1            411            105          516            21.317829   
2            979            239         1218            20.607553   
3            936            238         1174            21.720613   
4           1764            386         2150            18.744186   
..           ...            ...          ...                  ...   
511         1610            394         2004            20.409182   
512         5287           1156         6443            18.469657   
513         5497           1256         6753            19.917074   
514         2980            711         3691            19.642373   
515         3040            729         3769            20.164500   

    ObesityRiskCategory  
0              Moderate  
1                  High  
2                  High  
3                  High  
4              Moderate  
..                  ...  
511                High  
512            Moderate  
513            Moderate  
514            Moderate  
515                High  

[516 rows x 13 columns]

# **reduksi data (Data Reduction)**

**menghapus kolom** Mengapa dihapus?



Dataset lebih ringkas

Mengurangi ukuran file

Fokus pada informasi utama

In [ ]:
columns_to_drop = ["ValidCounts", "UnvalidCounts", "TotalCounts"]
df_cleaned = df.drop(columns=columns_to_drop)

In [ ]:
df

SchoolYear               NameHospital     Sex  EpiUnderweight  \
0          2001                    Borders  Female               2   
1          2001                    Borders    Male               5   
2          2001                    Lothian  Female              12   
3          2001                    Lothian    Male              17   
4          2001                       Fife  Female              17   
..          ...                        ...     ...             ...   
511        2023                    Tayside    Male              15   
512        2023  Greater Glasgow and Clyde  Female              34   
513        2023  Greater Glasgow and Clyde    Male              89   
514        2023                Lanarkshire  Female              14   
515        2023                Lanarkshire    Male              31   

     EpiHealthyWeight  EpiOverweight  EpiObese  EpiOverweightAndObese  \
0                 334             54        28                     82   
1                 301             63        42                    105   
2                 728            133       106                    239   
3                 681            124       114                    238   
4                1361            231       155                    386   
..                ...            ...       ...                    ...   
511              1201            216       178                    394   
512              4097            595       561                   1156   
513              4152            615       641                   1256   
514              2255            381       330                    711   
515              2280            348       381                    729   

     ValidCounts  UnvalidCounts  TotalCounts  UnhealthyPercentage  \
0            418             82          500            16.800000   
1            411            105          516            21.317829   
2            979            239         1218            20.607553   
3            936            238         1174            21.720613   
4           1764            386         2150            18.744186   
..           ...            ...          ...                  ...   
511         1610            394         2004            20.409182   
512         5287           1156         6443            18.469657   
513         5497           1256         6753            19.917074   
514         2980            711         3691            19.642373   
515         3040            729         3769            20.164500   

    ObesityRiskCategory  
0              Moderate  
1                  High  
2                  High  
3                  High  
4              Moderate  
..                  ...  
511                High  
512            Moderate  
513            Moderate  
514            Moderate  
515                High  

[516 rows x 13 columns]

In [ ]:
df

SchoolYear               NameHospital     Sex  EpiUnderweight  \
0          2001                    Borders  Female               2   
1          2001                    Borders    Male               5   
2          2001                    Lothian  Female              12   
3          2001                    Lothian    Male              17   
4          2001                       Fife  Female              17   
..          ...                        ...     ...             ...   
511        2023                    Tayside    Male              15   
512        2023  Greater Glasgow and Clyde  Female              34   
513        2023  Greater Glasgow and Clyde    Male              89   
514        2023                Lanarkshire  Female              14   
515        2023                Lanarkshire    Male              31   

     EpiHealthyWeight  EpiOverweight  EpiObese  EpiOverweightAndObese  \
0                 334             54        28                     82   
1                 301             63        42                    105   
2                 728            133       106                    239   
3                 681            124       114                    238   
4                1361            231       155                    386   
..                ...            ...       ...                    ...   
511              1201            216       178                    394   
512              4097            595       561                   1156   
513              4152            615       641                   1256   
514              2255            381       330                    711   
515              2280            348       381                    729   

     ValidCounts  UnvalidCounts  TotalCounts  UnhealthyPercentage  \
0            418             82          500            16.800000   
1            411            105          516            21.317829   
2            979            239         1218            20.607553   
3            936            238         1174            21.720613   
4           1764            386         2150            18.744186   
..           ...            ...          ...                  ...   
511         1610            394         2004            20.409182   
512         5287           1156         6443            18.469657   
513         5497           1256         6753            19.917074   
514         2980            711         3691            19.642373   
515         3040            729         3769            20.164500   

    ObesityRiskCategory  
0              Moderate  
1                  High  
2                  High  
3                  High  
4              Moderate  
..                  ...  
511                High  
512            Moderate  
513            Moderate  
514            Moderate  
515                High  

[516 rows x 13 columns]